### PR TITLE
[OJS]Pkp/pkp lib#2178 add nav menu manage

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

File: classes/services/NavigationMenuService.inc.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+ */
+
+namespace OJS\Services;
+
+class NavigationMenuService extends \PKP\Services\PKPNavigationMenuService {
+
+	/**
+	 * Initialize hooks for extending PKPSubmissionService
+	 */
+    public function __construct() {
```
Comment (NateWr) [inline classes/services/NavigationMenuService.inc.php]: Indentation is off here.

File: classes/services/NavigationMenuService.inc.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+		\HookRegistry::register('NavigationMenus::itemTypes', array($this, 'getMenuItemTypesCallback'));
+		\HookRegistry::register('NavigationMenus::displaySettings', array($this, 'getDisplayStatusCallback'));
+	}
+
+	/**
+	 * Return all default navigationMenuItemTypes.
+	 * @param $types array of types
```
Comment (NateWr) [inline classes/services/NavigationMenuService.inc.php]: Update description and `@param` docs.

File: classes/services/NavigationMenuService.inc.php [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+			),
+		);
+
+		$types = array_merge($types, $ojsTypes);
+	}
+
+	/**
+	 * Callback for display menu item functionallity
+	 */
+	function getDisplayStatusCallback($hookName, $args) {
```
Comment (NateWr) [inline classes/services/NavigationMenuService.inc.php]: Document function params.

File: classes/services/NavigationMenuService.inc.php [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+		$request = \Application::getRequest();
+		$dispatcher = $request->getDispatcher();
+
+		$isUserLoggedIn = \Validation::isLoggedIn();
+		$isUserLoggedInAs = \Validation::isLoggedInAs();
+		$context = $request->getContext();
+		$currentUser = $request->getUser();
+
+		$contextId = $context ? $context->getId() : CONTEXT_ID_NONE;
```
Comment (NateWr) [inline classes/services/NavigationMenuService.inc.php]: `$currentUser` and `$contextId` are not used in this method.

File: classes/services/NavigationMenuService.inc.php [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+		// Transform an item title if the title includes a {$variable}
```
Comment (NateWr) [inline classes/services/NavigationMenuService.inc.php]: Pull the title transformation code out into a method in `PKPNavigationMenuService` so it's not duplicated here.

File: classes/services/NavigationMenuService.inc.php [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+		$templateMgr = \TemplateManager::getManager(\Application::getRequest());
+		$title = $navigationMenuItem->getLocalizedTitle();
+		$prefix = '{$';
+		$postfix = '}';
+
+		$titleRepl = $title;
```
Comment (NateWr) [inline classes/services/NavigationMenuService.inc.php]: This line is unnecessary.

File: classes/services/NavigationMenuService.inc.php (hunk 1 above)
Comment (defstat) [reply classes/services/NavigationMenuService.inc.php]: OK

File: classes/services/NavigationMenuService.inc.php (hunk 2 above)
Comment (defstat) [reply classes/services/NavigationMenuService.inc.php]: OK

File: classes/services/NavigationMenuService.inc.php (hunk 3 above)
Comment (defstat) [reply classes/services/NavigationMenuService.inc.php]: OK

File: classes/services/NavigationMenuService.inc.php (hunk 4 above)
Comment (defstat) [reply classes/services/NavigationMenuService.inc.php]: OK

File: classes/services/NavigationMenuService.inc.php (hunk 6 above)
Comment (defstat) [reply classes/services/NavigationMenuService.inc.php]: Why?

File: classes/services/NavigationMenuService.inc.php (hunk 5 above)
Comment (defstat) [reply classes/services/NavigationMenuService.inc.php]: OK

File: classes/services/NavigationMenuService.inc.php (hunk 6 above)
Comment (NateWr) [reply classes/services/NavigationMenuService.inc.php]: You re-assign the variable on this line before it's used in any other way:

https://github.com/pkp/pkp-lib/pull/2831/files#diff-670d712d9da243cf8b3bab9244035f09R353

Comment (defstat) [reply classes/services/NavigationMenuService.inc.php]: OK

**Comment by NateWr** [comment]
Good! No further comments.

**Pull request merged**